demo-extended/build.gradle

```diff
@@ -29,6 +29,6 @@ android {
 
 dependencies {
     compile project(':library')
-    compile 'com.android.support:appcompat-v7:22.2.1'
-    compile 'com.android.support:recyclerview-v7:22.2.1'
+    compile 'com.android.support:appcompat-v7:25.2.0'
+    compile 'com.android.support:recyclerview-v7:25.2.0'
```

Comment:
these version updates could have been done in a separate PR and the review and merge of that would have been pretty fast - now they are stuck next to this other code that i imagine will be discussed at length. Maybe look at splitting this PR?